 #                     IBM APPLIED Data Science Capstone Project

#      Segmenting and Clustering Neighborhoods in Toronto


In [1]:
import numpy as np                        # library to handle data in a vectorized manner

import pandas as pd                       # library for data analsysis
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

import json                               # library to handle JSON files

!conda install -c conda-forge geopy --yes           # uncomment this line if you haven't completed the Foursquare API lab
from geopy.geocoders import Nominatim               # convert an address into latitude and longitude values

import requests                                    # library to handle requests
from pandas.io.json import json_normalize                 # tranform JSON file into a pandas dataframe

# Matplotlib and associated plotting modules
import matplotlib.cm as cm
import matplotlib.colors as colors

# import k-means from clustering stage
from sklearn.cluster import KMeans

!conda install -c conda-forge folium=0.5.0 --yes          # uncomment this line if you haven't completed the Foursquare API lab
import folium                                             # map rendering library

print('Libraries imported.')

Solving environment: done

## Package Plan ##

  environment location: /opt/conda/envs/Python36

  added / updated specs: 
    - geopy


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    certifi-2019.11.28         |           py36_0         149 KB  conda-forge
    geopy-1.21.0               |             py_0          58 KB  conda-forge
    openssl-1.1.1d             |       h516909a_0         2.1 MB  conda-forge
    ca-certificates-2019.11.28 |       hecc5488_0         145 KB  conda-forge
    geographiclib-1.50         |             py_0          34 KB  conda-forge
    ------------------------------------------------------------
                                           Total:         2.5 MB

The following NEW packages will be INSTALLED:

    geographiclib:   1.50-py_0         conda-forge
    geopy:           1.21.0-py_0       conda-forge

The following packages will be UPDATED:

    ca-

#   WebScraping Toronto Boroughs & Pin-Codes.


Install Beautiful Soup Package & Continue....

Extract the WikiTable with 287 rows.

In [2]:
from bs4 import BeautifulSoup
import requests
import csv

source = requests.get('https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M').text

soup = BeautifulSoup(source, 'lxml')

table=soup.table.tbody

table_df=pd.DataFrame()

for string in table.stripped_strings:
    table_df=table_df.append([string])

Borough_df=pd.DataFrame(table_df.to_numpy().reshape(-1,3))
Borough_df.columns = Borough_df.iloc[0]
Borough_df=Borough_df.drop(0).reset_index(drop=True)
Borough_df

0   Postcode           Borough  \
0        M1A      Not assigned   
1        M2A      Not assigned   
2        M3A        North York   
3        M4A        North York   
4        M5A  Downtown Toronto   
5        M6A        North York   
6        M6A        North York   
7        M7A  Downtown Toronto   
8        M8A      Not assigned   
9        M9A         Etobicoke   
10       M1B       Scarborough   
11       M1B       Scarborough   
12       M2B      Not assigned   
13       M3B        North York   
14       M4B         East York   
15       M4B         East York   
16       M5B  Downtown Toronto   
17       M5B  Downtown Toronto   
18       M6B        North York   
19       M7B      Not assigned   
20       M8B      Not assigned   
21       M9B         Etobicoke   
22       M9B         Etobicoke   
23       M9B         Etobicoke   
24       M9B         Etobicoke   
25       M9B         Etobicoke   
26       M1C       Scarborough   
27       M1C       Scarborough   
28       M1C       Scarborough   
29       M2C      Not assigned   
30       M3C        North York   
31       M3C        North York   
32       M4C         East York   
33       M5C  Downtown Toronto   
34       M6C              York   
35       M7C      Not assigned   
36       M8C      Not assigned   
37       M9C         Etobicoke   
38       M9C         Etobicoke   
39       M9C         Etobicoke   
40       M9C         Etobicoke   
41       M1E       Scarborough   
42       M1E       Scarborough   
43       M1E       Scarborough   
44       M2E      Not assigned   
45       M3E      Not assigned   
46       M4E      East Toronto   
47       M5E  Downtown Toronto   
48       M6E              York   
49       M7E      Not assigned   
50       M8E      Not assigned   
51       M9E      Not assigned   
52       M1G       Scarborough   
53       M2G      Not assigned   
54       M3G      Not assigned   
55       M4G         East York   
56       M5G  Downtown Toronto   
57       M6G  Downtown Toronto   
58       M7G      Not assigned   
59       M8G      Not assigned   
60       M9G      Not assigned   
61       M1H       Scarborough   
62       M2H        North York   
63       M3H        North York   
64       M3H        North York   
65       M3H        North York   
66       M4H         East York   
67       M5H  Downtown Toronto   
68       M5H  Downtown Toronto   
69       M5H  Downtown Toronto   
70       M6H      West Toronto   
71       M6H      West Toronto   
72       M7H      Not assigned   
73       M8H      Not assigned   
74       M9H      Not assigned   
75       M1J       Scarborough   
76       M2J        North York   
77       M2J        North York   
78       M2J        North York   
79       M3J        North York   
80       M3J        North York   
81       M4J         East York   
82       M5J  Downtown Toronto   
83       M5J  Downtown Toronto   
84       M5J  Downtown Toronto   
85       M6J      West Toronto   
86       M6J      West Toronto   
87       M7J      Not assigned   
88       M8J      Not assigned   
89       M9J      Not assigned   
90       M1K       Scarborough   
91       M1K       Scarborough   
92       M1K       Scarborough   
93       M2K        North York   
94       M3K        North York   
95       M3K        North York   
96       M4K      East Toronto   
97       M4K      East Toronto   
98       M5K  Downtown Toronto   
99       M5K  Downtown Toronto   
100      M6K      West Toronto   
101      M6K      West Toronto   
102      M6K      West Toronto   
103      M7K      Not assigned   
104      M8K      Not assigned   
105      M9K      Not assigned   
106      M1L       Scarborough   
107      M1L       Scarborough   
108      M1L       Scarborough   
109      M2L        North York   
110      M2L        North York   
111      M3L        North York   
112      M4L      East Toronto   
113      M4L      East Toronto   
114      M5L  Downtown Toronto   
115      M5L  Downtown Toronto   
116      M6L        No

### Now that we have the required table. Let us remove 'Not assigned' [210 rows remain]  and Data Wrangling
After Wrangling, we are left with 103 unique postal codes...!

In [3]:
#Remove 'NA' Borough, if Neighbourhood="NA", use Borough value
Borough_df=Borough_df[Borough_df['Borough']!='Not assigned'].reset_index(drop=True)

Borough_df=Borough_df.replace('Not assigned',np.NaN)
Borough_df["COL3"] = Borough_df["Neighbourhood"].fillna(Borough_df["Borough"])
Borough_df=Borough_df.drop(['Neighbourhood'],axis=1).reset_index(drop=True)
Borough_df. rename (columns={'COL3':'Neighbourhood'}, inplace=True)

#Group by Postcode & Borough with Concatenation of Neighbourhood Vales

Grouped_df=Borough_df.groupby(['Postcode','Borough'],as_index=False).agg(', '.join)
print(Grouped_df.head())

print('\n \n The dataframe has {} boroughs \n and \n {} ROWS .'.format(
        len(Grouped_df['Borough'].unique()),
        Grouped_df.shape[0]))

0 Postcode      Borough                           Neighbourhood
0      M1B  Scarborough                          Rouge, Malvern
1      M1C  Scarborough  Highland Creek, Rouge Hill, Port Union
2      M1E  Scarborough       Guildwood, Morningside, West Hill
3      M1G  Scarborough                                  Woburn
4      M1H  Scarborough                               Cedarbrae

 
 The dataframe has 10 boroughs 
 and 
 103 ROWS .


### Fetch Location data from the given link & Join with boroughs data [Tor_df]  

In [4]:
loc_df = pd.read_csv ('http://cocl.us/Geospatial_data')
loc_df.head()

Postal Code   Latitude  Longitude
0         M1B  43.806686 -79.194353
1         M1C  43.784535 -79.160497
2         M1E  43.763573 -79.188711
3         M1G  43.770992 -79.216917
4         M1H  43.773136 -79.239476

In [5]:
Tor_df = pd.merge(Grouped_df, loc_df, left_on='Postcode', right_on='Postal Code', how='left')
Tor_df=Tor_df.drop(['Postal Code'],axis=1).reset_index(drop=True)
Tor_df

Postcode           Borough  \
0        M1B       Scarborough   
1        M1C       Scarborough   
2        M1E       Scarborough   
3        M1G       Scarborough   
4        M1H       Scarborough   
5        M1J       Scarborough   
6        M1K       Scarborough   
7        M1L       Scarborough   
8        M1M       Scarborough   
9        M1N       Scarborough   
10       M1P       Scarborough   
11       M1R       Scarborough   
12       M1S       Scarborough   
13       M1T       Scarborough   
14       M1V       Scarborough   
15       M1W       Scarborough   
16       M1X       Scarborough   
17       M2H        North York   
18       M2J        North York   
19       M2K        North York   
20       M2L        North York   
21       M2M        North York   
22       M2N        North York   
23       M2P        North York   
24       M2R        North York   
25       M3A        North York   
26       M3B        North York   
27       M3C        North York   
28       M3H        North York   
29       M3J        North York   
30       M3K        North York   
31       M3L        North York   
32       M3M        North York   
33       M3N        North York   
34       M4A        North York   
35       M4B         East York   
36       M4C         East York   
37       M4E      East Toronto   
38       M4G         East York   
39       M4H         East York   
40       M4J         East York   
41       M4K      East Toronto   
42       M4L      East Toronto   
43       M4M      East Toronto   
44       M4N   Central Toronto   
45       M4P   Central Toronto   
46       M4R   Central Toronto   
47       M4S   Central Toronto   
48       M4T   Central Toronto   
49       M4V   Central Toronto   
50       M4W  Downtown Toronto   
51       M4X  Downtown Toronto   
52       M4Y  Downtown Toronto   
53       M5A  Downtown Toronto   
54       M5B  Downtown Toronto   
55       M5C  Downtown Toronto   
56       M5E  Downtown Toronto   
57       M5G  Downtown Toronto   
58       M5H  Downtown Toronto   
59       M5J  Downtown Toronto   
60       M5K  Downtown Toronto   
61       M5L  Downtown Toronto   
62       M5M        North York   
63       M5N   Central Toronto   
64       M5P   Central Toronto   
65       M5R   Central Toronto   
66       M5S  Downtown Toronto   
67       M5T  Downtown Toronto   
68       M5V  Downtown Toronto   
69       M5W  Downtown Toronto   
70       M5X  Downtown Toronto   
71       M6A        North York   
72       M6B        North York   
73       M6C              York   
74       M6E              York   
75       M6G  Downtown Toronto   
76       M6H      West Toronto   
77       M6J      West Toronto   
78       M6K      West Toronto   
79       M6L        North York   
80       M6M              York   
81       M6N              York   
82       M6P      West Toronto   
83       M6R      West Toronto   
84       M6S      West Toronto   
85       M7A  Downtown Toronto   
86       M7R       Mississauga   
87       M7Y      East Toronto   
88       M8V         Etobicoke   
89       M8W         Etobicoke   
90       M8X         Etobicoke   
91       M8Y         Etobicoke   
92       M8Z         Etobicoke   
93       M9A         Etobicoke   
94       M9B         Etobicoke   
95       M9C         Etobicoke   
96       M9L        North York   
97       M9M        North York   
98       M9N              York   
99       M9P         Etobicoke   
100      M9R         Etobicoke   
101      M9V         Etobicoke   
102      M9W         Etobicoke   

                                         Neighbourhood   Latitude  Longitude  
0                                       Rouge, Malvern  43.806686 -79.194353  
1               Highland Creek, Rouge Hill, Port Union  43.784535 -79.160497  
2                    Guildwood, Morningside, West Hill  43.763573 -79.188711  
3                                               Woburn  43.770992 -79.216917  
4                                            Cedarbrae  43.773136 -79.23

## Create a map of Toronto Neighbourhoods using FOLIUM

Use Geopy to get Location of Toronto Neighbourhoods

Create Neighbourhood Map of Toronto

In [6]:
address = 'Toronto, Canada'

geolocator = Nominatim(user_agent="foursquare_agent")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print('The geograpical coordinate of Toronto City are {}, {}.'.format(latitude, longitude))

The geograpical coordinate of Toronto City are 43.653963, -79.387207.


## Select only Toronto Boroughs

In [7]:
NH_df=Tor_df[Tor_df['Borough'].str.contains('Toronto')].sort_values (by = 'Borough').drop(['Postcode'],axis=1).reset_index(drop=True)
NH_df

Borough                                      Neighbourhood  \
0    Central Toronto                The Annex, North Midtown, Yorkville   
1    Central Toronto                Forest Hill North, Forest Hill West   
2    Central Toronto                                      Lawrence Park   
3    Central Toronto                                   Davisville North   
4    Central Toronto                                 North Toronto West   
5    Central Toronto                                         Davisville   
6    Central Toronto                        Moore Park, Summerhill East   
7    Central Toronto  Deer Park, Forest Hill SE, Rathnelly, South Hi...   
8    Central Toronto                                           Roselawn   
9   Downtown Toronto  Harbourfront East, Toronto Islands, Union Station   
10  Downtown Toronto                     Harbord, University of Toronto   
11  Downtown Toronto                     Commerce Court, Victoria Hotel   
12  Downtown Toronto           Design Exchange, Toronto Dominion Centre   
13  Downtown Toronto                                       Queen's Park   
14  Downtown Toronto                           Adelaide, King, Richmond   
15  Downtown Toronto                                 Central Bay Street   
16  Downtown Toronto                                       Harbourfront   
17  Downtown Toronto                                     St. James Town   
18  Downtown Toronto                           Ryerson, Garden District   
19  Downtown Toronto          Chinatown, Grange Park, Kensington Market   
20  Downtown Toronto                               Church and Wellesley   
21  Downtown Toronto                        Cabbagetown, St. James Town   
22  Downtown Toronto                                           Rosedale   
23  Downtown Toronto                    Stn A PO Boxes 25 The Esplanade   
24  Downtown Toronto             First Canadian Place, Underground city   
25  Downtown Toronto                                           Christie   
26  Downtown Toronto                                        Berczy Park   
27  Downtown Toronto  CN Tower, Bathurst Quay, Island airport, Harbo...   
28      East Toronto                                        The Beaches   
29      East Toronto                     The Beaches West, India Bazaar   
30      East Toronto                       The Danforth West, Riverdale   
31      East Toronto                                    Studio District   
32      East Toronto  Business Reply Mail Processing Centre 969 Eastern   
33      West Toronto                       Dovercourt Village, Dufferin   
34      West Toronto                           Little Portugal, Trinity   
35      West Toronto       Brockton, Exhibition Place, Parkdale Village   
36      West Toronto                      High Park, The Junction South   
37      West Toronto                             Parkdale, Roncesvalles   
38      West Toronto                                 Runnymede, Swansea   

     Latitude  Longitude  
0   43.672710 -79.405678  
1   43.696948 -79.411307  
2   43.728020 -79.388790  
3   43.712751 -79.390197  
4   43.715383 -79.405678  
5   43.704324 -79.388790  
6   43.689574 -79.383160  
7   43.686412 -79.400049  
8   43.711695 -79.416936  
9   43.640816 -79.381752  
10  43.662696 -79.400049  
11  43.648198 -79.379817  
12  43.647177 -79.381576  
13  43.662301 -79.389494  
14  43.650571 -79.384568  
15  43.657952 -79.387383  
16  43.654260 -79.360636  
17  43.651494 -79.375418  
18  43.657162 -79.378937  
19  43.653206 -79.400049  
20  43.665860 -79.383160  
21  43.667967 -79.367675  
22  43.679563 -79.377529  
23  43.646435 -79.374846  
24  43.648429 -79.382280  
25  43.669542 -79.422564  
26  43.644771 -79.373306  
27  43.628947 -79.394420  
28  43.676357 -79.293031  
29  43.668999 -79.315572  
30  43.679557 -79.352188  
31  43.659526 -79.340923  
32  43.662744 -79.321558  
33  43.669005 -79.442259  
34  43.647927 -79.419750  
35  43.636847 -79.428191  
36  43.661608

In [8]:
# create map of Toronto with Boroughs using latitude and longitude values
map_tor = folium.Map(location=[latitude, longitude], zoom_start=11)

# add markers to map
for lat, lng, borough, neighborhood in zip(NH_df['Latitude'], NH_df['Longitude'], NH_df['Borough'], NH_df['Neighbourhood']):
    label = '{}'.format(neighborhood)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_tor)  
    
map_tor

Enter the Fourquare credentials

In [9]:
CLIENT_ID = 'GQYZTHDW4FELZGYMP0DIQXVDFGXO5MMAVZATGRXRFKCPJV4X' # your Foursquare ID
CLIENT_SECRET = 'IWPETGS0XZIYMQBDYOZO2QRWDE0UJFHOBRA1YPTL1PK3F4SL' # your Foursquare Secret
VERSION = '20190202'
LIMIT = 100
print('Your credentails have been entered')


Your credentails have been entered


##### TEST if we can obtain data from Fourquare:

Get Restaurants data from FourSquare Toronto City are 43.653963, -79.387207.

In [10]:
search_query = 'Restaurant'  #['Restaurant','School','Bar','Medical','Shop']
radius = 500


url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&v={}&query={}&radius={}&limit={}'\
.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude, VERSION, search_query, radius, LIMIT)

results = requests.get(url).json()


In [11]:
# assign relevant part of JSON to venues
venues = results['response']['venues']

# tranform venues into a dataframe
rest_df = json_normalize(venues)
rest_df.head()

categories  hasPerk  \
0  [{'id': '4bf58dd8d48988d14e941735', 'name': 'A...    False   
1  [{'id': '52af3b773cf9994f4e043c03', 'name': 'S...    False   
2  [{'id': '4bf58dd8d48988d145941735', 'name': 'C...    False   
3  [{'id': '4bf58dd8d48988d1c4941735', 'name': 'R...    False   
4  [{'id': '4bf58dd8d48988d14a941735', 'name': 'V...    False   

                         id          location.address location.cc  \
0  4ad4c05ff964a52048f720e3       110 Chestnut Street          CA   
1  5750b013498e755287c6de97        988 Baldwin Street          CA   
2  4b2027b5f964a520f82d24e3           195 Dundas St W          CA   
3  50ca02c0245f2d4aa8c2b313  145 Richmond Street West          CA   
4  4c476d6719fde21e32410876         179 Dundas St. W.          CA   

  location.city location.country location.crossStreet  location.distance  \
0       Toronto           Canada                  NaN                145   
1       Toronto           Canada                  NaN                573   
2       Toronto           Canada    at University Ave                107   
3       Toronto           Canada       Hilton Toronto                458   
4       Toronto           Canada          at Chestnut                140   

                           location.formattedAddress  \
0  [110 Chestnut Street, Toronto ON M5G 1R3, Canada]   
1           [988 Baldwin Street, Toronto ON, Canada]   
2  [195 Dundas St W (at University Ave), Toronto ...   
3  [145 Richmond Street West (Hilton Toronto), To...   
4  [179 Dundas St. W. (at Chestnut), Toronto ON M...   

                             location.labeledLatLngs  location.lat  \
0  [{'label': 'display', 'lat': 43.65488413420439...     43.654884   
1  [{'label': 'display', 'lat': 43.655874, 'lng':...     43.655874   
2  [{'label': 'display', 'lat': 43.65492521335936...     43.654925   
3  [{'label': 'display', 'lat': 43.65000998764964...     43.650010   
4  [{'label': 'display', 'lat': 43.65506808, 'lng...     43.655068   

   location.lng location.postalCode location.state  \
0    -79.385931             M5G 1R3             ON   
1    -79.393826                 NaN             ON   
2    -79.387089             M5G 1C7             ON   
3    -79.385608             M5H 2L2             ON   
4    -79.386375                 M5G             ON   

                                 name    referralId venuePage.id  
0     Hemispheres Restaurant & Bistro  v-1582716437          NaN  
1  Some Time BBQ Grill Restaurant 碳烤屋  v-1582716437          NaN  
2       Hong Shing Chinese Restaurant  v-1582716437     60327598  
3                   Tundra Restaurant  v-1582716437          NaN  
4                     Cali Restaurant  v-1582716437          NaN

In [12]:
rest_df=rest_df[['name','location.lat','location.lng']]
rest_df['name'].nunique()

40

###### There are 40 restaurants within 500m of Toronto City Center.

###### Get data for Number of Restaurants, Bars, Schools, Medical Centres & Shops for all the Neighbourhoods of TORONTO.

###### For this repetitive process, Create a user-defined function to Get Venues & Store them in Data Frame !

In [13]:
def getNearbyVenues(names, latitudes, longitudes, radius=500):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&v={}&query={}&radius={}&limit={}'\
.format(CLIENT_ID, CLIENT_SECRET, lat, lng, VERSION, search_query, radius, LIMIT)

            
        # make the GET request
        results = requests.get(url).json()
        venues = results['response']['venues']

        # tranform venues into a dataframe
        #rest_df = json_normalize(venues)
        #rest_df=rest_df[['name']]
        
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name,
            lat,
            lng,
            v['name']) for v in venues])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Neighbourhood', 
                  'Neighbourhood Latitude', 
                  'Neighbourhood Longitude', 
                  'Venue']
    
    return(nearby_venues)

##### Use the user-defined function to get Data for all Neighbourhoods.

In [14]:
Tor_venues = getNearbyVenues(names=NH_df['Neighbourhood'],
                                   latitudes=NH_df['Latitude'],
                                   longitudes=NH_df['Longitude']
                                  )
Tor_venues

Neighbourhood  \
0                  Forest Hill North, Forest Hill West   
1                  Forest Hill North, Forest Hill West   
2                                     Davisville North   
3                                     Davisville North   
4                                     Davisville North   
5                                     Davisville North   
6                                   North Toronto West   
7                                   North Toronto West   
8                                           Davisville   
9                                           Davisville   
10                                          Davisville   
11                                          Davisville   
12                                          Davisville   
13   Deer Park, Forest Hill SE, Rathnelly, South Hi...   
14   Deer Park, Forest Hill SE, Rathnelly, South Hi...   
15   Deer Park, Forest Hill SE, Rathnelly, South Hi...   
16   Deer Park, Forest Hill SE, Rathnelly, South Hi...   
17   Harbourfront East, Toronto Islands, Union Station   
18   Harbourfront East, Toronto Islands, Union Station   
19   Harbourfront East, Toronto Islands, Union Station   
20   Harbourfront East, Toronto Islands, Union Station   
21   Harbourfront East, Toronto Islands, Union Station   
22   Harbourfront East, Toronto Islands, Union Station   
23   Harbourfront East, Toronto Islands, Union Station   
24   Harbourfront East, Toronto Islands, Union Station   
25   Harbourfront East, Toronto Islands, Union Station   
26   Harbourfront East, Toronto Islands, Union Station   
27   Harbourfront East, Toronto Islands, Union Station   
28   Harbourfront East, Toronto Islands, Union Station   
29   Harbourfront East, Toronto Islands, Union Station   
30   Harbourfront East, Toronto Islands, Union Station   
31   Harbourfront East, Toronto Islands, Union Station   
32   Harbourfront East, Toronto Islands, Union Station   
33   Harbourfront East, Toronto Islands, Union Station   
34   Harbourfront East, Toronto Islands, Union Station   
35                      Harbord, University of Toronto   
36                      Harbord, University of Toronto   
37                      Harbord, University of Toronto   
38                      Harbord, University of Toronto   
39                      Commerce Court, Victoria Hotel   
40                      Commerce Court, Victoria Hotel   
41                      Commerce Court, Victoria Hotel   
42                      Commerce Court, Victoria Hotel   
43                      Commerce Court, Victoria Hotel   
44                      Commerce Court, Victoria Hotel   
45                      Commerce Court, Victoria Hotel   
46                      Commerce Court, Victoria Hotel   
47                      Commerce Court, Victoria Hotel   
48                      Commerce Court, Victoria Hotel   
49                      Commerce Court, Victoria Hotel   
50                      Commerce Court, Victoria Hotel   
51                      Commerce Court, Victoria Hotel   
52                      Commerce Court, Victoria Hotel   
53                      Commerce Court, Victoria Hotel   
54                      Commerce Court, Victoria Hotel   
55                      Commerce Court, Victoria Hotel   
56                      Commerce Court, Victoria Hotel   
57                      Commerce Court, Victoria Hotel   
58                      Commerce Court, Victoria Hotel   
59                      Commerce Court, Victoria Hotel   
60                      Commerce Court, Victoria Hotel   
61                      Commerce Court, Victoria Hotel   
62                      Commerce Court, Victoria Hotel   
63                      Commerce Court, Victoria Hotel   
64                      Commerce Court, Victoria Hotel   
65                      Commerce Court, Victoria Hotel   
66                      Commerce Court, Victoria Hotel   
67                      Commerce Court, Victoria Hotel   
68                      Commerce Court,

In [15]:
Tor_Group=Tor_venues.groupby(['Neighbourhood'],as_index=False).count()

Tor_Group

Neighbourhood  Neighbourhood Latitude  \
0                            Adelaide, King, Richmond                      42   
1                                         Berczy Park                      13   
2        Brockton, Exhibition Place, Parkdale Village                       4   
3                         Cabbagetown, St. James Town                       8   
4                                  Central Bay Street                      41   
5           Chinatown, Grange Park, Kensington Market                      46   
6                                            Christie                       3   
7                                Church and Wellesley                      18   
8                      Commerce Court, Victoria Hotel                      46   
9                                          Davisville                       5   
10                                   Davisville North                       4   
11  Deer Park, Forest Hill SE, Rathnelly, South Hi...                       4   
12           Design Exchange, Toronto Dominion Centre                      39   
13                       Dovercourt Village, Dufferin                       1   
14             First Canadian Place, Underground city                      42   
15                Forest Hill North, Forest Hill West                       2   
16                     Harbord, University of Toronto                       4   
17                                       Harbourfront                       6   
18  Harbourfront East, Toronto Islands, Union Station                      18   
19                      High Park, The Junction South                       1   
20                           Little Portugal, Trinity                       8   
21                                 North Toronto West                       2   
22                             Parkdale, Roncesvalles                       4   
23                                       Queen's Park                       9   
24                                 Runnymede, Swansea                       7   
25                           Ryerson, Garden District                      46   
26                                     St. James Town                      41   
27                    Stn A PO Boxes 25 The Esplanade                      28   
28                                    Studio District                       2   
29                                        The Beaches                       1   
30                     The Beaches West, India Bazaar                       4   
31                       The Danforth West, Riverdale                      11   

    Neighbourhood Longitude  Venue  
0                        42     42  
1                        13     13  
2                         4      4  
3                         8      8  
4                        41     41  
5                        46     46  
6                         3      3  
7                        18     18  
8                        46     46  
9                         5      5  
10                        4      4  
11                        4      4  
12                       39     39  
13                        1      1  
14                       42     42  
15                        2      2  
16                        4      4  
17                        6      6  
18                       18     18  
19                        1      1  
20                        8      8  
21                        2      2  
22                        4      4  
23                        9      9  
24                        7      7  
25                       46     46  
26                       41     41  
27                       28     28  
28                        2      2  
29                        1      1  
30                        4      4  
31                       11     11